types/subcall.go

```diff
@@ -0,0 +1,59 @@
+package types
+
+import "encoding/json"
+
+// SubMsg wraps a CosmosMsg with some metadata for handling replies (ID) and optionally
+// limiting the gas usage (GasLimit)
+type SubMsg struct {
+	ID       uint64    `json:"id"`
+	Msg      CosmosMsg `json:"msg"`
+	GasLimit *uint64   `json:"gas_limit,omitempty"`
```

Comment:
The JSON schema says this is `null` or an integer:

```
    "SubMsg_for_CustomMsg": {
      "description": "A sub-message that will guarantee a subcall_response callback on success or error Note on error the subcall will revert any partial state changes due to this message, but not revert any state changes in the calling contract (that must be done in the subcall_response entry point)",
      "type": "object",
      "required": [
        "id",
        "msg"
      ],
      "properties": {
        "gas_limit": {
          "type": [
            "integer",
            "null"
          ],
          "format": "uint64",
          "minimum": 0.0
        },
        "id": {
          "type": "integer",
          "format": "uint64",
          "minimum": 0.0
        },
        "msg": {
          "$ref": "#/definitions/CosmosMsg_for_CustomMsg"
        }
      }
    },
```

Are you sure it works with `*uint64` and `omitempty`?

Reply:
It is not a required field. This works fine

Reply:
Alright, and it only needs to work from Rust to Go. Then we receive a number of `null` in Go.